server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreLockManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import java.io.IOException;
+
+/**
+ * An Interface that defines Index Level Remote Store MD Lock Manager.
+ * @opensearch.internal
+ */
+public interface RemoteStoreLockManager {
+    /**
+     *
+     * @param lockInfo lock info instance for which we need to acquire lock.
+     * @throws IOException throws exception if lock is not present.
```

Comment:
The description of the exception may not be correct. Please check.

What happens,
- if the object on which we are trying to acquire lock is not present.
- if object is present but acquiring lock failed
- if the lock is already present.